main/ble_hidd_demo_main.c

```diff
@@ -496,15 +522,90 @@ void uart_parse_command (uint8_t character, struct cmdBuf * cmdBuffer)
+                    
+                /////// Mouse
                 } else if (cmdBuffer->buf[1] == 0x03) {  // mouse report
-                    esp_hidd_send_mouse_value(hid_conn_id,cmdBuffer->buf[2],cmdBuffer->buf[3],cmdBuffer->buf[4],cmdBuffer->buf[5]);
+					///If we are using the rate limiter, check for the last transmission of a HID command.
+					/// Note: For mouse reports, updated values are either accumulated or discarded.
+					///  Especially with the FLipMouse, many small reports with small movements are sent.
+					///  So either we discard them (easy way) or accumulate the axis' values & sent the final value next possible occasion.
+					#if CONFIG_MODULE_USERATELIMITER
+						//check interval, just to be sure use the absolute value of the subtraction.
+						if(abs(esp_timer_get_time() - cmdBuffer->lastTimeHIDCommandSent) > timeIntervalHIDCommand)
+						{
+							#if CONFIG_MODULE_RATELIMITERACCUMULATE
+								//get for all 3 relative axis the current value, limit if necessary
+								//subtract from accumulated value the amount which is sent now.
+								int8_t current[3];
+								for(uint8_t i = 0; i<3; i++)
+								{
+									//add current values
+									cmdBuffer->accumValues[i] += cmdBuffer->buf[i+3];
+									//limit accumulated value, if we hit the upper limit, there 
+									// is something going wrong...
+									if(cmdBuffer->accumValues[i] > 6000) cmdBuffer->accumValues[i] = 6000;
+									if(cmdBuffer->accumValues[i] < -6000) cmdBuffer->accumValues[i] = -6000;
+									
+									//limit values on upper&lower boundaries
+									if(cmdBuffer->accumValues[i] > 127) {
+										current[i] = 127; cmdBuffer->accumValues[i] -= 127;
+									} else if(cmdBuffer->accumValues[i] < -127) {
+										current[i] = -127; cmdBuffer->accumValues[i] += 127;
+									} else {
+										current[i] = cmdBuffer->accumValues[i];
+										cmdBuffer->accumValues[i] = 0;
+									}
+									esp_hidd_send_mouse_value(hid_conn_id,cmdBuffer->buf[2],current[0],current[1],current[2]);
+								}
+							#else /* CONFIG_MODULE_RATELIMITERACCUMULATE */
+								esp_hidd_send_mouse_value(hid_conn_id,cmdBuffer->buf[2],cmdBuffer->buf[3],cmdBuffer->buf[4],cmdBuffer->buf[5]);
+							#endif /* CONFIG_MODULE_RATELIMITERACCUMULATE */
+							cmdBuffer->lastTimeHIDCommandSent = esp_timer_get_time();
+						} else {
+							#if CONFIG_MODULE_RATELIMITERACCUMULATE
+								for(uint8_t i = 0; i<3; i++)
+								{
+									//add current values
+									cmdBuffer->accumValues[i] += cmdBuffer->buf[i+3];
+									//limit accumulated value, if we hit the upper limit, there 
+									// is something going wrong...
+									if(cmdBuffer->accumValues[i] > 6000) cmdBuffer->accumValues[i] = 6000;
+									if(cmdBuffer->accumValues[i] < -6000) cmdBuffer->accumValues[i] = -6000;
+								}
+							#else /* CONFIG_MODULE_RATELIMITERACCUMULATE */
+								ESP_LOGI(EXT_UART_TAG,"Rate limit: discard mouse report");
```

Comment:
I think this might create a whole lot of log messages in case of slower rate limits.
maybe limit that output, too ;-)

Reply:
I changed this output to log level verbose, which is off by default.